examples/main/mnist.hs

```diff
@@ -11,15 +11,19 @@ import           Control.Monad
 import           Control.Monad.Random
 import           Control.Monad.Trans.Except
 
-import qualified Data.Attoparsec.Text as A
-import           Data.List ( foldl' )
-import           Data.Semigroup ( (<>) )
-import qualified Data.Text as T
-import qualified Data.Text.IO as T
-import qualified Data.Vector.Storable as V
+import qualified Data.Attoparsec.Text          as A
```

Comment:
We should avoid semaniticly null changes like that.

Reply:
Sure. I’m afraid that I ran the code through `Brittany ` to make it easier to work on (for me). I don’t think that it’s worth merging as it still doesn’t work.

Reply:
Must say, I am not a fan of `Brittany` or any of the other auto format tools. I am however a huge fan of minimal diffs that avoid semantically null changes :).

Feel free to close this PR.

Reply:
Sure, OK.  It's be good to get the canonical example in the field working, and understand why this version eats memory (which might also be documented).   The code for reading the lecun files is easy enough to reinstate.

Reply:
If you have any suggestions documentation improvements would be more than happy to have them. The code to handle the raw Lecun files may also be appropriate, but it is far easier to review a PR when the diff is minimal.

Reply:
Erik, I'm not seeing how fuller documentation is going to address the underlying issue, namely that the example doesn't work (either in my variant, with the lecun data, or with a csv file).  The mnist.hs file isn't large and so, of course, I could redo it against the current base, with minimal changes (although my OCD likes my language features and imports in alphabetical order :-) ) but there is clearly something awry that we should try to fix first.

A couple of the other examples work OK (just using `cabal run feedforward` , etc).  I haven't tried the shakespeare (which I assume also requires training input in a particular format)

Reply:
Ah never mind. I thought you had it working.